Release/src/http/client/http_client_asio.cpp

```diff
@@ -1659,7 +1663,66 @@ pplx::task<http_response> asio_client::propagate(http_request request)
     // Asynchronously send the response with the HTTP client implementation.
     this->async_send_request(context);
 
-    return result_task;
+    return result_task.then(std::bind(&asio_client::maybe_redirect,
+                                      this,
+                                      std::placeholders::_1,
+                                      request,
+                                      std::vector<uri>()));
+}
+
+pplx::task<http_response> asio_client::maybe_redirect(http_response response, http_request request, std::vector<uri> redirected_uris) {
+    static const status_code found_status_code = 302;
```

Comment:
Shouldnt we also react to:

- 301 Moved Permanently
- 307 Temporary Redirect
- 308 Permanent Redirect